teps/0079-tekton-catalog-support-tiers.md

```diff
@@ -517,7 +523,117 @@ TODO
+    - name: say-world
+      image: alpine
+      script: echo 'world'
+```
+
+If the `image` value (or part of the value) is specified by `params`, the default value of the `params` (if provided) are collected and used to run the security report. For example, `gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/git-init:v0.40.2` is extracted from the following `task`:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: git-clone
+...
+spec:
+...
+  params:
+    - name: gitInitImage
+      description: The image providing the git-init binary that this Task runs.
+      type: string
+      default: "gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/git-init:v0.40.2"
+  steps:
+    - name: clone
+      image: "$(params.gitInitImage)"
+      script: |
+        #!/usr/bin/env sh
+        ...
+```
+
+Please note that it is presumed that the `image` field references the correct `param` and a valid container image string is formed after the substitution. If wrong values are specified in the `step.image` (either a wrong `param` pointer or invalid container image value), the resource itself is invalid and therefore no vulnerability report will be generated for this step.
+
+Similarly, `bash:latest` and `alpine` are extracted from the following `pipeline`:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: demo-pipeline
+...
+spec:
+  tasks:
+  - name: say-hello
+    taskSpec:
+      steps:
+      - name: say-hello-step
+        image: bash:latest
+        script: echo 'hello'
+  - name: say-world
+    taskSpec:
+      steps:
+      - name: say-world-step
+        image: alpine
+        script: echo 'world'
+```
+
+If a `pipeline` resource contains `pipelineTask` specified by `taskRef` for example:
+
+```yaml
+tasks:
+  - name: build-image
+    taskRef:
+      name: build-and-push
+```
+the container images used in the `pipelineTask` (i.e. `build-and-push`) will **not** be included in the security report.
+
+
+#### Design Evaluation
+
+This design builds on the existing infrastructure and the security report UI design of the Artifact Hub. A significant amount of engineering effort can be saved by leveraging the out-of-box features provided by the Artifact Hub.
+
+#### Alternatives
+##### 1. Store Container Images in Metadata File
+
+The Artifact Hub provides a [generic solution](https://artifacthub.io/docs/topics/security_report/#coredns-plugins-keda-scalers-keptn-integrations-opa-policies-and-tinkerbell-actions) to specify the container images in a dedicated [`artifacthub-pkg.yml`](https://github.com/artifacthub/hub/blob/master/docs/metadata/artifacthub-pkg.yml#L13) metadata file that can be parsed by the Artifact Hub. While this solution can be easily opted in for Tekton resources, it is an extra cost for Tekton users to maintain the metadata file. Also, we cannot guarantee that the images being scanned are the images actually used in the resource if the metadata file is outdated.
```

Comment:
I'm not sure why this is an alternative and not complementary to the proposed solution.
AFAIK the images must be parsed out of Tekton resources and stored somewhere. Wouldn't that be the `artifacthub-pkg.yml` file, or where will they be stored otherwise?

I don't understand why this would be an extra effort for Tekton users, since the metadata file can be generated from the resources, it doesn't have to be maintained manually.

Reply:
Yeah, like I explained above, the container images are stored in the DB not `artifacthub-pkg.yml`.

The `artifacthub-pkg.yml` file is not auto-generated based on the resources, it has to be prepared by users manually (and whatever images provided in the `artifacthub-pkg.yml` by user will be sent for the vulnerability scanning).